src/sentry/middleware/ratelimit.py

```diff
@@ -22,6 +22,8 @@
 class RatelimitMiddleware(MiddlewareMixin):
     """Middleware that applies a rate limit to every endpoint."""
 
+    rate_limit_check_dict: dict | None = None
```

Comment:
I think this could be a better type signature. You could either:

1. Make this a dataclass (unless there's a specific reason this needs to be a dict).
2. use [TypedDict](https://adamj.eu/tech/2021/05/10/python-type-hints-how-to-use-typeddict/)

Either way, it should be clear what values this configuration is expecting

Reply:
Also, is this holding local state of some sort? How is that supposed to work? Does it get cleared on every request?

Reply:
The rate limit is stored as a property of this middleware. Before a request gets processed, it'll hit process_view first. This will set the rate limit check metadata. After the view function returns, process_response will add the rate limit information to the headers.

Each request has it's own middleware stack so the values are unique per request.

Reply:
> Each request has it's own middleware stack so the values are unique per request.

Got it, that's what I was making sure of